### PR TITLE
Resource reload hooks

### PR DESCRIPTION
**Pull request by asiekierka**
@Falkreon

### CODE REVIEW

File: src/main/java/net/fabricmc/fabric/api/client/itemgroup/FabricItemGroupBuilder.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public static FabricItemGroupBuilder create(Identifier identifier) {
 
 	/**
 	 *
-	 * This is used to add an icon to to the item group
+	 * This is used to addReloadListener an icon to to the item group
```
[the remark this answers is not in the document]
Comment (asiekierka) [reply src/main/java/net/fabricmc/fabric/api/client/itemgroup/FabricItemGroupBuilder.java]: Oh it is

**Comment by asiekierka** [comment]
As this seems very non-controversial, I'd like to ask:

* Is there any scenario where you would want to load *before* something (say, between tags and recipes -> before recipes), or *before* everything? I'm honestly of the opinion you're better off mixing the particular provider then...

**Comment by falkreon** [comment]
...or would be better solved by having some kind of well-defined priority order on the resource packs themselves, most likely. Yeah, before isn't super useful here.

**Comment by asiekierka** [comment]
One final thing: I'm not sure if I shouldn't change the IdentifiableResourceReloadListener interface into a builder pattern taking in normal ResourceReloadListeners + adding identifier/dependency (incl. optional dependencies for ordering - speaking of which, is that even necessary?) information, for further extensibility.

Not sure if it's likely we're ever gonna extend this, is all.

**Pull request closed without merging**